app/Http/Controllers/BillsController.php

```diff
@@ -202,11 +204,40 @@ public function sendMailBill($id)
     {
         $bills = Bills::find($id);
         $emailAddress = $bills->paymentReference->vendor->email;
-        Mail::to($emailAddress)->queue(new MailVendorBill);
+
+        Mail::to($emailAddress)->queue(new MailVendorBill ($bills));
         
-        return redirect()->route('bills.bill.index')->with('success', 'Email has been sent!');
+        return redirect()->route('bills.bills.index')->with('success', 'Email has been sent!');
     }
 
+    public function sendMailPurchaseOrder($id)
+    {
+        $purchaseOrders = PurchaseOrders::find($id);
+        $emailAddress = $purchaseOrders->paymentReference->vendor->email;
+        
+        Mail::to($emailAddress)->queue(new MailVendorPurchaseOrder ($purchaseOrders));
+        
+        return redirect()->route('bills.bills.index')->with('success', 'Email has been sent!');
+    }
+
+    // Print
+    public function printBill($id)
+    {
+        $bills = Bills::find($id);
+
+        $pdf = PDF::loadView('vendors.bills.print', compact('bills'));
```

Comment:
I think singular form of naming should be used here. Not a big deal for now, but it helps minimizing confusions.